language/move-prover/spec-lang/src/env.rs

```diff
@@ -423,9 +429,34 @@ impl GlobalEnv {
             .any(|d| d.severity >= Severity::Error)
     }
 
-    /// Writes accumulated diagnostics to writer.
+    /// Returns true if diagnostics have warning severity or worse.
+    pub fn has_warnings(&self) -> bool {
+        self.diags
+            .borrow()
+            .iter()
+            .any(|d| d.severity >= Severity::Warning)
+    }
+
+    /// Writes accumulated errors to writer.
     pub fn report_errors<W: WriteColor>(&self, writer: &mut W) {
-        for diag in self.diags.borrow().iter() {
+        for diag in self
+            .diags
+            .borrow()
+            .iter()
+            .filter(|d| d.severity >= Severity::Error)
+        {
+            emit(writer, &Config::default(), &self.source_files, diag).expect("emit must not fail");
+        }
+    }
+
+    /// Writes accumulated warnings to writer.
```

Comment:
This actually seems to write warnings and errors. Do we want to change the doc or the impl?